### DISCUSSION TITLE
How to use Grafana plugin with MQTT datasource?

### DISCUSSION THREAD

**Discussion by guenter-ms** [hassio-addons/addon-grafana#348] [Q&A]
Hi,
I'm using Grafana on HA and it works fine.
Now I like to use the grafana mqtt datasource and configured a connection to an mqtt broker on a separate machine under nodered.
The connection test in the mqtt datasource is ok:
MQTT Connected
When trying to explore a mqtt topic I get no results / no data
What goes wrong?

Do I need to follow this procedure? > https://grafana.com/docs/grafana/latest/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-streaming-data-source-plugin/

Thanks for any help

See screenshots:

![Bildschirmfoto 2023-08-17 um 17 48 41](https://github.com/hassio-addons/addon-grafana/assets/6253315/c11d1e21-c268-47ec-8d07-bff3e3497e7b)

![Bildschirmfoto 2023-08-17 um 17 49 13](https://github.com/hassio-addons/addon-grafana/assets/6253315/cb9c086c-7b54-4568-b2ec-9be0b30734aa)

![Bildschirmfoto 2023-08-17 um 17 50 21](https://github.com/hassio-addons/addon-grafana/assets/6253315/4dc956c9-01d9-492d-baab-57e3375dfac5)

![Bildschirmfoto 2023-08-17 um 17 55 50](https://github.com/hassio-addons/addon-grafana/assets/6253315/5c64a804-d696-4482-a3a0-e529d9186a5e)

**Comment by wesseli01** [comment]
Hi @guenter-ms,
Have you found a solution? I am fighting with the same problem with the addon in HA. If I am using the access from the grafana server outsinde of HA then I get the mqtt data over the mqtt connection source in grafana (with source HA ....tcp://XXXXX:1883  -> XXXXX: IP of HA) 
Kind regards, 
wesseli

@frenck , Could you support us, please? We would like to use the mqtt connection directly.

**Reply by guenter-ms** [reply]
i, not solved yet. See: https://github.com/grafana/grafana/issues/73582#issuecomment-1694254755
